tests/handlers/tests.py

```diff
@@ -38,7 +38,8 @@ def test_bad_path_info(self):
         environ['PATH_INFO'] = b'\xed' if six.PY2 else '\xed'
         handler = WSGIHandler()
         response = handler(environ, lambda *a, **k: None)
-        self.assertEqual(response.status_code, 400)
+        # The path of the request will be encoded to '/%ED'.
```

Comment:
I've briefly audited the `WSGIRequest` constructor, technically a `UnicodeDecodeError` is still possible from `get_script_name`. Replacing environ['PATH_INFO'] with `environ['SCRIPT_NAME']` should hit the same codepath.

Regarding removing the exception catching, that's for another ticket, but we need to check whether the RFCs expect us to repercent `SCRIPT_NAME` as well. If we did that, the exception catching could go.